From 0431b069c2b995643909cd74742108a03a47ab03 Mon Sep 17 00:00:00 2001
From: Jianxiong Pan <jianxiong.pan@amlogic.com>
Date: Wed, 7 Sep 2022 12:37:37 +0800
Subject: [PATCH 46/95] mm: fix slab info stat error. [2/2]

PD#SWPL-92714

Problem:
Slab:              54672 kB
SReclaimable:      54672 kB
SUnreclaim:            0 kB

Solution:
correct the slab info statistic.

Verify:
sc2_ah212.

Change-Id: Ic7ddd1125b65f42000f02710fc7c9b94805820f4
Signed-off-by: Jianxiong Pan <jianxiong.pan@amlogic.com>
---
 mm/slab_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/slab_common.c b/mm/slab_common.c
index acf5ff9c1422..84e884a3e882 100644
--- a/mm/slab_common.c
+++ b/mm/slab_common.c
@@ -978,7 +978,7 @@ void *kmalloc_order(size_t size, gfp_t flags, unsigned int order)
 		ret = page_address(page);
 	#ifdef CONFIG_AMLOGIC_MEMORY_EXTEND
 		mod_lruvec_page_state(page, NR_SLAB_UNRECLAIMABLE_B,
-				      PAGE_ALIGN(size) / PAGE_SIZE);
+				      PAGE_ALIGN(size));
 	#else
 		mod_lruvec_page_state(page, NR_SLAB_UNRECLAIMABLE_B,
 				      PAGE_SIZE << order);
-- 
2.25.1

